src/Raven.Client/Documents/Changes/EvictItemsFromCacheBasedOnChanges.cs

```diff
@@ -12,31 +12,51 @@
 
 namespace Raven.Client.Documents.Changes
 {
-    internal class EvictItemsFromCacheBasedOnChanges : IObserver<DocumentChange>, IObserver<IndexChange>, IDisposable
+    internal class EvictItemsFromCacheBasedOnChanges : IObserver<DocumentChange>, IObserver<IndexChange>, IObserver<AggressiveCacheUpdate>, IDisposable
     {
         private readonly string _databaseName;
-        private readonly IDatabaseChanges _changes;
-        private readonly IDisposable _documentsSubscription;
-        private readonly IDisposable _indexesSubscription;
+        private readonly DatabaseChanges _changes;
+        private IDisposable _documentsSubscription;
+        private IDisposable _indexesSubscription;
         private readonly RequestExecutor _requestExecutor;
+        private readonly Task _taskConnected;
+        private IDisposable _aggressiveCachingSubscription;
 
         public EvictItemsFromCacheBasedOnChanges(DocumentStore store, string databaseName)
         {
             _databaseName = databaseName;
-            _changes = store.Changes(databaseName);
             _requestExecutor = store.GetRequestExecutor(databaseName);
-            var docSub = _changes.ForAllDocuments();
-            _documentsSubscription = docSub.Subscribe(this);
-            var indexSub = _changes.ForAllIndexes();
-            _indexesSubscription = indexSub.Subscribe(this);
+            _changes = new DatabaseChanges(_requestExecutor, databaseName, null, null);
```

Comment:
can we use named parameters here? really hard to know what is here without opening the VS.

That being said, double-checking here, because nodeTag is `null` so we will pick preferred node here. Looks fine for me.

Reply:
Done